hw/ip/hmac/README.md

```diff
@@ -15,41 +15,42 @@ See that document for integration overview within the broader OpenTitan top leve
 
 ## Features
 
-- HMAC supporting multiple digest sizes: SHA-2 256/384/512 hashing algorithm
-- HMAC-SHA-2 and unkeyed SHA-2 dual mode
+- Dual modes: HMAC based on SHA-2 and unkeyed SHA-2
+- Multiple digest sizes supported (for both modes): SHA-2 256/384/512 hashing algorithm
 - Configurable key length 128/256/384/512/1024-bit secret key for HMAC mode
+- Support for context switching (via saving and restoring) across multiple message streams
 - 32 x 32-bit message buffer
 
 ## Description
 
 [sha256-spec]: https://csrc.nist.gov/publications/detail/fips/180/4/final
 
 The HMAC module is a [SHA-2][sha256-spec] hash-based authentication code generator to check the integrity of an incoming message and a signature signed with the same secret key.
-It supports SHA-2 256/384/512 and 128/256/384/512/1024-bit keys in HMAC mode, so long as the key length does not exceed the block size of the configured SHA-2 mode, i.e., 1024-bit keys are not supported for SHA-2 256 where the block size is 512-bit.
+It supports SHA-2 256/384/512 and 128/256/384/512/1024-bit keys in HMAC mode, so long as the key length does not exceed the block size of the configured SHA-2 digest size, i.e., 1024-bit keys are not supported for SHA-2 256 where the block size is 512-bit.
 It generates a different authentication code with the same message if the secret key is different.
 
 This HMAC implementation is not hardened against side channel or fault injection attacks.
 It is meant purely for hashing acceleration.
 If hardened MAC operations are required, users should use either [KMAC](../kmac/README.md) or a software implementation.
 
-The 1024-bit secret key is written in [`KEY_0`](doc/registers.md#key) to [`KEY_31`](doc/registers.md#key), and the key length relevant to the HMAC operation is configured in [`CFG.key_length`].
-For example, to use a 256-bit secret key, [`CFG.key_length`] should be configured to 0x02 and then only secret key registers [`KEY_0`](doc/registers.md#key) to [`KEY_7`] are read and relevant for the HMAC operation.
+The secret key is written in [`KEY_0`](doc/registers.md#key) to [`KEY_31`](doc/registers.md#key), and the key length relevant to the HMAC operation is configured in [`CFG.key_length`].
```

Comment:
``[`CFG.key_length`]`` doesn't result in an implicit link, unfortunately. `(doc/registers.md#cfg--key_length)` needs to be appended. This problem also occurs in existing lines in this file; may I ask you to please fix them too in this PR?

Reply:
In this context, I would bundle "`KEY_0` to `KEY_31`" into a single link (and similar for "`DIGEST_0` to `DIGEST_15`" etc)